File: app/src/main/java/com/hussain/podcastapp/database/EntryDao.java
package com.hussain.podcastapp.database;

import com.hussain.podcastapp.model.Entry;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface EntryDao {

    @Query("SELECT * FROM entry")
    List<Entry> loadAllPodcasts();

    @Insert
    void insertPodcast(Entry movies);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertPodcastList(List<Entry> movies);

    @Query("SELECT * FROM entry WHERE id = :id")
    Entry getPodcast(String id);

    @Query("DELETE FROM entry WHERE id = :id")
    void deletePodcast(String id);
}

File: app/src/main/java/com/hussain/podcastapp/database/EntryDao.kt
package com.hussain.podcastapp.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.hussain.podcastapp.model.Entry

@Dao
interface EntryDao {

    @Query("SELECT * FROM entry")
    fun loadAllPodcasts(): List<Entry>

    @Insert
    fun insertPodcast(movies: Entry)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertPodcastList(movies: List<Entry>)

    @Query("SELECT * FROM entry WHERE id = :id")
    fun getPodcast(id: String): Entry

    @Query("DELETE FROM entry WHERE id = :id")
    fun deletePodcast(id: String)
}
